src/newViews/Tagging/Export/NewAssessmentExport/AssessmentsExportSelection/index.tsx

```diff
@@ -0,0 +1,239 @@
+}
+
+const exportItems: ExportItem = {
+    assessment: 'assessment',
+    plannedAssessment: 'planned_assessment',
+};
+
+interface ExportTriggerResponse {
+    exportTriggered: number;
+}
+
+interface Props {
+    className?: string;
+    projectRole: {
+        exportPermissions?: {
+            'create_only_unprotected'?: boolean;
+        };
+    };
+    projectId: number;
+}
+
+function AssessmentsExportSelection(props: Props) {
+    const {
+        className,
+        projectId,
+        projectRole,
+    } = props;
+
+    const filterOnlyUnprotected = !!projectRole?.exportPermissions?.create_only_unprotected;
+    const [exportClass, setExportClass] = useState<string>();
+    const [queryTitle, setQueryTitle] = useState<string>();
+    const [exportItem, setExportItem] = useState<string>();
+    const [previewId, setPreviewId] = useState<number | undefined>(undefined);
+    const [isPreview, setIsPreview] = useState<boolean>(false);
+    const [selectedLeads, setSelectedLeads] = useState<number[]>([]);
+    const [selectAll, setSelectAll] = useState<boolean>(true);
+    const [filterValues, setFilterValues] = useState<SourceEntryFilter>({});
+
+    const {
+        pending: exportPending,
+        trigger: getExport,
+    } = useLazyRequest<ExportTriggerResponse, unknown>({
+        url: 'server://export-trigger/',
+        method: 'POST',
+        body: ctx => ({ filters: ctx }),
+        onSuccess: (response) => {
+            if (isPreview) {
+                setPreviewId(response.exportTriggered);
+            } else if (exportItem === exportItems.assessment) {
+                notify.send({
+                    title: _ts('export', 'headerExport'),
+                    type: notify.type.SUCCESS,
+                    message: _ts('export', 'exportStartedNotifyMessage'),
+                    duration: 15000,
+                });
+            }
+
+            setExportClass(undefined);
+        },
+        failureHeader: _ts('export', 'headerExport'),
+    });
+
+    const startExport = useCallback((preview: boolean, item: string) => {
+        const filters = {
+            project: projectId,
+            include_leads: !selectAll,
+            lead: selectedLeads,
+
+
+            export_type: 'excel',
+            // NOTE: export_type for 'word' and 'pdf' is report so, we need to differentiate
+            pdf: false,
+
+            // entry or assessment
+            export_item: item,
+
+            // temporary or permanent
+            is_preview: preview,
+            ...filterValues,
+        };
+
+        setIsPreview(preview);
+        setExportItem(item);
+
+        const newExportClass = (
+            (preview && EXPORT_CLASS.assessmentPreview)
+            || (item === exportItems.plannedAssessment && EXPORT_CLASS.plannedAryExport)
+            || (item === exportItems.assessment && EXPORT_CLASS.assessmentExport)
+            || undefined
+        );
+        setExportClass(newExportClass);
```

Comment:
All of these can be derived from the export context. So, we may not need to store these